tests/unit/lms/models/grading_info_test.py

```diff
@@ -57,6 +57,44 @@ def test_it_enforces_uniqueness_constraint(
         ):
             db_session.flush()
 
+    def test_repr(self, grading_info):
+        assert repr(grading_info) == (
+            "GradingInfo("
+            "id=None, "
+            "created=None, "
+            "updated=None, "
+            "lis_result_sourcedid='result_sourcedid', "
+            "lis_outcome_service_url='https://somewhere.else', "
+            "oauth_consumer_key='test_consumer_key', "
+            "user_id='339483948', "
+            "context_id='random context', "
+            "resource_link_id='random resource link id', "
+            "tool_consumer_info_product_family_code='MyFakeLTITool', "
+            "h_username='ltiuser1', "
+            "h_display_name='My Fake LTI User')"
+        )
+
+    def test_repr_is_valid_python(self, grading_info):
+        deserialized_grading_info = eval(repr(grading_info))  # pylint:disable=eval-used
+
+        for attr in (
+            "id",
+            "created",
+            "updated",
+            "lis_result_sourcedid",
+            "lis_outcome_service_url",
+            "oauth_consumer_key",
+            "user_id",
+            "context_id",
+            "resource_link_id",
+            "tool_consumer_info_product_family_code",
+            "h_username",
+            "h_display_name",
+        ):
+            assert getattr(deserialized_grading_info, attr) == getattr(
+                grading_info, attr
+            )
```

Comment:
Did you find this property handy during debugging? I feel a bit strange about writing a test to maintain a property we don't make use of. I think this property is generally broken enough in the wild that it's not really a thing.

These tests would probably disappear to be replaced with something shorter in `BaseClass` anyway if you liked that idea, so maybe it doesn't really matter either way.

Reply:
The fact that it's valid Python? No that's not really useful but I just think it's the right style

Reply:
(See the Python docs: [<q>If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value</q>](https://docs.python.org/3/reference/datamodel.html?highlight=__str__#object.__repr__). It's supposed to be <q>information-rich and unambiguous</q> for debugging purposes and a Python expression is just a good style for achieving that. The fact that you could literally copy-paste-execute it in the debugger to create another instance, might come in handy sometimes but isn't really the point.)

Reply:
I know, I was talking about the tests not the style. In general we shouldn't write tests to preserve properties and behavior we don't use. It's pain if you make a change which you know is fine but the tests fail anyway because they fix some prorperty the product doesn't actually rely on.

But as I said, it's so much shorter when done generally that I don't think it matters here, and if you have to have coverage of `__repr__` in your tests, unless it only states it returns a string (might be a good idea actually), it's going to be sensitive to the contents anyway.